transmitter.go

```diff
@@ -243,21 +184,13 @@ func (t *transmitter) check(p pdu.PDU, n int, err error) (closing bool) {
 }
 
 // low level writing
-func (t *transmitter) write(v []byte) (n int, err error) {
-	hasTimeout := t.settings.Timeout > 0
-
-	if hasTimeout {
-		err = t.conn.SetWriteTimeout(t.settings.Timeout)
+func (t *transmitable) write(v []byte) (n int, err error) {
+	if t.settings.WriteTimeout > 0 {
+		err = t.conn.SetWriteTimeout(t.settings.WriteTimeout)
 	}
 
 	if err == nil {
-		if n, err = t.conn.Write(v); err != nil &&
-			n == 0 &&
-			hasTimeout &&
-			t.conn.SetWriteTimeout(t.settings.Timeout<<1) == nil {
-			// retry again with double timeout
-			n, err = t.conn.Write(v)
-		}
+		n, err = t.conn.Write(v)
```

Comment:
Just for my knowledge ... why do you remove retrying ?

Reply:
I refresh my mind if I should retry. 
- In case of broken connection or SMSC fail handling
- SMPP not support multiplexing

Due to those reasons, retrying won't help in sending PDU to SMSC successfully.

Reply:
:+1: